ws-slave.js

```diff
@@ -219,6 +176,9 @@ wss.on('connection', function (ws) {
     ws.removeAllListeners('close');
     ws.removeAllListeners('open');
     ws.removeAllListeners('message');
+    if (contextsLeft <= 0) {
+      noble.stopScanning();
```

Comment:
this could probably go to notifyScanRelevantEvent, since a context appearing/disappearing is scan-relevant.  It'll still end up stopping scanning, but keeps everything moving through that chokepoint for scan/don't-scan decisions.